convert2rhel/toolopts.py

```diff
@@ -97,40 +127,63 @@ def _get_argparser():
+            r"  convert2rhel {analyze}"
             "\n\n"
             "*WARNING* The tool needs to be run under the root user"
         )
         return argparse.ArgumentParser(
-            prog="convert2rhel",
-            conflict_handler="resolve",
-            usage=usage,
-            add_help=False,
+            conflict_handler="resolve", usage=usage, formatter_class=argparse.ArgumentDefaultsHelpFormatter
         )
 
-    def _register_options(self):
-        """Prescribe what command line options the tool accepts."""
-        self._parser.add_argument(
-            "-h",
-            "--help",
-            action="help",
-            help="Show help message and exit.",
+    def _register_commands(self):
+        """ """
+        subparsers = self._parser.add_subparsers(title="Subcommands", dest="command")
+        self._analyze_parser = subparsers.add_parser(
+            "analyze",
+            help="Run all Convert2RHEL initial checks up until the "
+            " Point of no Return (PONR) and generate a report with the findings."
+            " A rollback is initiated after the checks to put the system back"
+            " in the original state.",
+            parents=[self._shared_options_parser],
+            formatter_class=argparse.ArgumentDefaultsHelpFormatter,
         )
-        self._parser.add_argument(
+        self._convert_parser = subparsers.add_parser(
+            "convert",
+            help="Convert the system.",
+            parents=[self._shared_options_parser],
+            formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+        )
+
+    def _register_parent_options(self, parser):
+        """Prescribe what command line parent options the tool accepts."""
+        parser.add_argument(
             "--version",
             action="version",
             version=__version__,
             help="Show convert2rhel version and exit.",
         )
-        self._parser.add_argument(
+        parser.add_argument(
             "--debug",
             action="store_true",
             help="Print traceback in case of an abnormal exit and messages that could help find an issue.",
         )
-        # Importing here instead of on top of the file to avoid cyclic dependency
-        from convert2rhel.systeminfo import POST_RPM_VA_LOG_FILENAME, PRE_RPM_VA_LOG_FILENAME
 
+    def _register_options(self):
+        """Prescribe what command line options the tool accepts."""
+        self._parser.add_argument(
+            "-h",
+            "--help",
+            action="help",
+            help="Show help message and exit.",
+        )
+        # TODO: check if this option is suited to be a shared option
         self._parser.add_argument(
+            "--config_file",
+            action="append",
+            help="The configuration file is an optional way to safely pass either a user password or an activation key",
+        )
```

Comment:
Thinking about it right now, I guess that this could be a shared option, since it can be used for both conversions and analyze... I guess that some integration tests could fail because of this, no?

Reply:
There is a shared option for config file under subscription manager options, this is a duplicate which will be removed